### ISSUE TITLE
Force field loaded from nested child directory

### ISSUE THREAD

**Issue by SimonBoothroyd** [openforcefield/openff-toolkit#899]
**Describe the bug**

It seems like #816 introduces a bug whereby a `ForceField` can inadvertently be loaded from a nested child of the current working directory. This can presumably lead to subtle issues whereby the user may think they are loading a force field in their current working directory, but if they forgot to create / copy a file there, load a force field from a child directory with the same name but not be the intended force field.

This issue is related to #898 

**To Reproduce**

```
import os

from openff.toolkit.typing.engines.smirnoff import ForceField

nested_directory = os.path.join("a", "b", "c")
os.makedirs(nested_directory, exist_ok=True)

# Create a FF in a nested directory
ForceField("openff-1.3.0.offxml").to_file(
    os.path.join(nested_directory, "force-field.offxml")
)

# Check that the file does not exist in the current working directory.
assert not os.path.isfile("force-field.offxml")

print(ForceField("force-field.offxml"))
```

**Output**

```
<openff.toolkit.typing.engines.smirnoff.forcefield.ForceField object at 0x7fa44f45acd0>
```

**Computing environment (please complete the following information):**
 - Operating system: OSX
 - Output of running `conda list` 

```
openff-toolkit            0.9.1              pyhd8ed1ab_1    conda-forge
openff-toolkit-base       0.9.1              pyhd8ed1ab_1    conda-forge
python                    3.9.2           h2502468_0_cpython    conda-forge
```

**Additional context**

**Comment by j-wags** [comment]
@dfhahn had a problem with this as the root cause today.

**Comment by dfhahn** [comment]
@j-wags, @SimonBoothroyd thanks for the pointer. Yes, I encountered a problem which is related to this:

If there is a nested child directory, which has the same name as the force field to be loaded, `ForceField` will try to read in the directory, which leads to an error.

**To reproduce**

```
import os
from openff.toolkit.typing.engines.smirnoff import ForceField

nested_directory = os.path.join('a', 'b', 'forcefield.offxml')
os.makedirs(nested_directory)

ForceField('forcefield.offxml')
```

**Output**

```
OSError: Source a/b/forcefield.offxml could not be read. If this is a file, ensure that the path is correct.
If the file is present, ensure it is in a known SMIRNOFF encoding.
Valid formats are: ['XML']
Parsing failed with the following error:
not well-formed (invalid token): line 1, column 0
```

**Computing environment**

- OS: Linux
- `conda list`
```
openff-forcefields        2.0.0rc.1          pyh6c4a22f_0    conda-forge
openff-toolkit            0.9.2              pyhd8ed1ab_1    conda-forge
openff-toolkit-base       0.9.2              pyhd8ed1ab_1    conda-forge
```

**Comment by mattwthompson** [comment]
I'm not sure why I used this `glob` pattern: https://github.com/openforcefield/openff-toolkit/blob/85d1c75d7b9611f959bcaf335497d0e8e80fb647/openff/toolkit/typing/engines/smirnoff/forcefield.py#L1111

but it's definitely the source of these issues. From the [docs](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob):

> The “\*\*” pattern means “this directory and all subdirectories, recursively”. In other words, it enables recursive globbing:

so it makes sense that files in subdirectories are inadvertently being found.

**Comment by mattwthompson** [comment]
Should be fixed by #1002 - please re-open if not!
